### ISSUE TITLE
minikube profile should have it's own kubeconfig file

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#4669]
Currently, each minikube profile adds itself to the main kubeconfig (`~/.kube/config` ) as a context. 

it would be nice that on the top of adding itself to the main kube config, to add separate file.
like `~/.kube/minikube-profile1 `

this will be super useful when doing operations in parallel on multiple minikube profiles. and also will pave the way for support for minikube multi cluster.

currently if we want to do operations on two nodes at the same time, we have to set the kubeclt current-context, and there could be a race condition !

but if we have a separate kube-configfile for each profile we can pass as environment variable.
so we we could do operations like this:

```
KUBECONFIG="~/.kube/minikube-profile1"  kubectl apply -f example1_node1.yaml 
KUBECONFIG="~/.kube/minikube-profile2" kubectl apply -f example2_node2.yaml 
```

I could implement this myself if I have time but I would be happy to guide and review anyone who wants to pick this up !

while one might aruge we could do, `kubectl --context profile1` not all the kubectl commands take the --context option.

what do you guys think ? @tstromberg @afbjorklund

**Comment by medyagh** [comment]
Won't do
